bears/general/AnnotationBear.py

```diff
@@ -1,6 +1,81 @@
 from coalib.parsing.StringProcessing.Core import (search_for,
                                                   search_in_between,
                                                   unescaped_search_in_between)
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.results.SourceRange import SourceRange
+
+
+class AnnotationBear(LocalBear):
+
+    def run(self, filename, file, language: str, language_family: str):
+        """
+        Finds out all the positions of strings and comments in a file.
+        The Bear searches for valid comments and strings and yields their
+        ranges as SourceRange objects in HiddenResults.
+
+        :param language:        Language to be whose annotations are to be
+                                searched.
+        :param language_family: Language family whose annotations are to be
+                                searche.
+        :return:                HiddenResult with content conatining
+                                a tuple of tuples of the form
+                                (annotation_start, annotation_end).
+        """
+        text = ''.join(file)
+        lang_dict = LanguageDefinition(language, language_family)
+
+        # Strings
+        strings = dict(lang_dict["string_delimiters"])
+        strings.update(lang_dict["multiline_string_delimiters"])
+        strings_found = find_start_end(text, strings, escape=True)
```

Comment:
Why don't we directly return a SourceRange/SourcePosition or TextRange/TextPosition directly ? ?
We already have functions like `overlaps()` in SourceRange and TextRange and that can be used directly ?